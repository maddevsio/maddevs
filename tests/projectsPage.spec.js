import { mount, createLocalVue } from '@vue/test-utils'
import ProjectsPage from '@/pages/projects'
import Vuelidate from 'vuelidate'

const localVue = createLocalVue()

localVue.use(Vuelidate)

describe('projects page', () => {
  let wrapper

  beforeEach(() => {
    global.$nuxt = {
      $route: {
        name: null,
      },
    }
    wrapper = mount(ProjectsPage, {
      localVue,
      stubs: ['nuxt-link'],
    })
  })

  // ------ IMPORTANT ----- //
  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  // --------------------- //

  it('sets the correctly data', () => {
    expect(wrapper.vm.$data.title).toEqual('Mad Devs Portfolio: Key Clients, Case Studies and Open Source Projects')
    expect(wrapper.vm.$data.description).toEqual(
      'Check out our software engineering and mobile app development projects for companies from transportation, logistic, edtech, cloudtech, security, advertising, finance, and other industries.',
    )
    expect(wrapper.vm.$data.ogUrl).toEqual('https://maddevs.io/projects/')
  })
})
