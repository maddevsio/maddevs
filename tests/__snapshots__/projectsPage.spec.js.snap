// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projects page renders correctly 1`] = `
<main
  class="main"
>
  <div
    class="banner__banner-container banner-container"
  >
    <div
      class="banner__banner-content banner-content"
    >
      <div
        class="banner__main-title-wrapper main-title-wrapper container"
      >
        <h1
          class="banner__banner-main_title banner-main_title"
        >
          
        Check out our 
          <br />
           
          <span
            class="banner__main-title-white_part main-title-white_part"
          >
            
          50+
          
            <br
              class="banner__small-screen-break"
            />
            
          implemented
          
            <br />
            
          projects!
        
          </span>
        </h1>
      </div>
       
      <div
        class="navigation-wrap container navigation-wrap-"
      >
        <nav
          class="navigation-list navigation-list-"
        >
          <nuxtlink-stub
            class="navigation-item"
            to="#case-studies"
          >
            
      Case studies
    
          </nuxtlink-stub>
          <nuxtlink-stub
            class="navigation-item"
            to="#customers"
          >
            
      Our customers
    
          </nuxtlink-stub>
          <nuxtlink-stub
            class="navigation-item"
            to="#open-source"
          >
            
      Open-source pet projects
    
          </nuxtlink-stub>
           
        </nav>
      </div>
    </div>
  </div>
   
  <section
    class="case-studies case-studies--projects"
    id="case-studies"
  >
    <div
      class="container"
    >
      <div>
        <h2
          class="case-studies__main-title case-studies__main-title-projects"
        >
          
        Case 
          <br />
          
        Studies
      
        </h2>
         
        <div
          class="case-studies__fire-icon"
        />
      </div>
    </div>
     
    <div
      class="container"
    >
      <section
        class="cases-list"
      >
        <div
          class="cases-list_item cases-list_item--full cases-list_veeqo"
          data-testid="test-case-card"
        >
          <nuxtlink-stub
            class="cases-list_item-link"
            disabled=""
            tag="a"
            to="/case-studies/veeqo/"
          >
            <video
              class="cases-list_item-video media_lazy"
              loop="loop"
              muted="muted"
              poster="img.jpg"
            >
              <source
                data-src="img.jpg"
                type="video/mp4"
              />
              
      Your browser does not support the video tag.
    
            </video>
             
            <div
              class="cases-list_item-info"
            >
              <img
                alt="Veeqo logo"
                height="23"
                src="img.jpg"
                width="105"
              />
               
              <span>
                Logistics
              </span>
               
              <h3>
                
        Optimization for Veeqo
      
              </h3>
               
              <p>
                Our DevOps team optimized the infrastructure for Veeqo, an inventory and shipping e-commerce platform, and made it highly efficient, smooth, and cost-effective for our partners.
              </p>
               
              <nuxtlink-stub
                to="/case-studies/veeqo/"
              >
                
        Explore
      
              </nuxtlink-stub>
            </div>
          </nuxtlink-stub>
        </div>
        <div
          class="cases-list_item cases-list_item--big cases-list_itc"
          data-testid="test-case-card"
        >
          <nuxtlink-stub
            class="cases-list_item-link"
            disabled=""
            tag="a"
            to="/case-studies/R4TCA-web-application/"
          >
            <video
              class="cases-list_item-video media_lazy"
              loop="loop"
              muted="muted"
              poster="img.jpg"
            >
              <source
                data-src="img.jpg"
                type="video/mp4"
              />
              
      Your browser does not support the video tag.
    
            </video>
             
            <div
              class="cases-list_item-info"
            >
              <img
                alt="ITC logo"
                height="23"
                src="img.jpg"
                width="105"
              />
               
              <span>
                Digital transformation
              </span>
               
              <h3>
                
        Ready4Trade Central Asia
      
              </h3>
               
              <p>
                A web app customized for the delivery of export management coaching missions that contributes to the development of intra-regional and international trade in Central Asia
              </p>
               
              <nuxtlink-stub
                to="/case-studies/R4TCA-web-application/"
              >
                
        Explore
      
              </nuxtlink-stub>
            </div>
          </nuxtlink-stub>
        </div>
        <div
          class="cases-list_item cases-list_item--small cases-list_yourcast"
          data-testid="test-case-card"
        >
          <nuxtlink-stub
            class="cases-list_item-link"
            disabled=""
            tag="a"
            to="/case-studies/yourcast/"
          >
            <video
              class="cases-list_item-video media_lazy"
              loop="loop"
              muted="muted"
              poster="img.jpg"
            >
              <source
                data-src="img.jpg"
                type="video/mp4"
              />
              
      Your browser does not support the video tag.
    
            </video>
             
            <div
              class="cases-list_item-info"
            >
              <img
                alt="Yourcast logo"
                height="25"
                src="img.jpg"
                width="120"
              />
               
              <span>
                Content Streaming
              </span>
               
              <h3>
                
        The Evolution of Yourcast.TV
      
              </h3>
               
              <p>
                Mad Devs developed a secure and private video streaming service and live video broadcasting cinema to provide entertainment experience to isolated groups feeling homesick.
              </p>
               
              <nuxtlink-stub
                to="/case-studies/yourcast/"
              >
                
        Explore
      
              </nuxtlink-stub>
            </div>
          </nuxtlink-stub>
        </div>
        <div
          class="cases-list_item cases-list_item--full cases-list_sjmc"
          data-testid="test-case-card"
        >
          <nuxtlink-stub
            class="cases-list_item-link"
            disabled=""
            tag="a"
            to="/case-studies/sir-john-monash-centre/"
          >
            <video
              class="cases-list_item-video media_lazy"
              loop="loop"
              muted="muted"
              poster="img.jpg"
            >
              <source
                data-src="img.jpg"
                type="video/mp4"
              />
              
      Your browser does not support the video tag.
    
            </video>
             
            <div
              class="cases-list_item-info"
            >
              <img
                alt="SJMC logo"
                height="60"
                src="img.jpg"
                width="133"
              />
               
              <span>
                BYOD
              </span>
               
              <h3>
                
        Sir John Monash Centre
      
              </h3>
               
              <p>
                Mad Devs engineers helped Sir John Monash Centre to enhance and maintain the BYOD multimedia technology.
              </p>
               
              <nuxtlink-stub
                to="/case-studies/sir-john-monash-centre/"
              >
                
        Explore
      
              </nuxtlink-stub>
            </div>
          </nuxtlink-stub>
        </div>
      </section>
       
      <nuxtlink-stub
        class="cases-list_see-more"
        to="/case-studies/"
      >
        <span>
          See More
        </span>
      </nuxtlink-stub>
    </div>
  </section>
   
  <section
    class="customers"
    id="customers"
  >
    <div
      class="container"
    >
      <div
        class="customers__text-content"
      >
        <h2
          class="customers__main-title"
        >
          
        Our valued customers
      
        </h2>
         
        <p
          class="customers__description"
        >
          
        Due to our NDAs and confidentiality requirements, we cannot disclose details for all of our customers. These
        are a few of the companies we work with.
      
        </p>
      </div>
       
      <div
        class="customers__customers-list"
      >
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="godee"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="guardrails"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="teacherly"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="triggermesh"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="clutch"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="eatigo"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="veeqo"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="admitad"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="iotsploit"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="sjmc"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="progressay"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="kickoico"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="nambafood"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="showmebishkek"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="mythril"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="seso"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="tengri"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="peklo"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="bandpay"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="nambataxi"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="alfa-bank"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="edumall"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="dzing"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="topica"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="terbine"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="silk-road"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="vms"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="ase"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
        <div
          class="icon-wrapper"
          data-testid="icon-wrapper"
          style="background-color: rgb(25, 26, 27);"
        >
          <img
            alt="iothub"
            class="icon-wrapper__icon media_lazy"
            data-src="[object Object]"
            data-testid="customer-icon"
            height="64"
            width="143"
          />
        </div>
      </div>
    </div>
  </section>
   
  <section
    class="open-source"
    id="open-source"
  >
    <div
      class="container"
    >
      <div
        class="open-source__top-text-content"
      >
        <h3
          class="open-source__main-title"
        >
          
        Our code - available in open source - is used by many other companies.
      
        </h3>
         
        <div
          class="open-source__github-wrap"
        >
          <p
            class="open-source__description"
          >
            
          70+ of 
            <span>
              Mad Devs'
            </span>
             pet projects have been shared with 
            <br
              class="open-source__descktop-break"
            />
            
          the community via
          
            <span
              class="open-source__mobile-icon-wrapper"
            >
              GitHub.
            
              <a
                class="open-source__github-icon-mobile"
                href="https://github.com/maddevsio"
                target="_blank"
              />
            </span>
          </p>
           
          <a
            class="open-source__github-icon"
            href="https://github.com/maddevsio"
            target="_blank"
          />
        </div>
      </div>
       
      <div
        class="open-source__projects"
      >
        <div
          class="open-source__projects-item"
        >
          <div
            class="open-source__project open-source__project-mad-location"
          >
            <a
              class="open-source__project-link"
              href="https://github.com/maddevsio/mad-location-manager"
              target="_blank"
            >
              <div
                class="open-source__content-wrap"
              >
                <div
                  class="open-source__icon open-source__mad-location"
                />
              </div>
               
              <p
                class="open-source__paragraph"
              >
                А library for GPS and accelerometer data "fusion" with a Kalman filter.
              </p>
            </a>
          </div>
        </div>
        <div
          class="open-source__projects-item"
        >
          <div
            class="open-source__project open-source__project-comedian"
          >
            <a
              class="open-source__project-link"
              href="https://github.com/maddevsio/comedian"
              target="_blank"
            >
              <div
                class="open-source__content-wrap"
              >
                <div
                  class="open-source__icon open-source__comedian"
                />
              </div>
               
              <p
                class="open-source__paragraph"
              >
                A 
                <span>
                  team-management
                </span>
                 system leveraging Slack and Telegram bot functionalities to enable remote standups and track daily performance.
              </p>
            </a>
          </div>
        </div>
        <div
          class="open-source__projects-item"
        >
          <div
            class="open-source__project open-source__project-ariadna"
          >
            <a
              class="open-source__project-link"
              href="https://github.com/maddevsio/ariadna"
              target="_blank"
            >
              <div
                class="open-source__content-wrap"
              >
                <div
                  class="open-source__icon open-source__ariadna"
                />
              </div>
               
              <p
                class="open-source__paragraph"
              >
                An 
                <span>
                  open-source
                </span>
                 geocoder, built on top of ElasticSearch, for fast geocoding and better search for CIS countries.
              </p>
            </a>
          </div>
        </div>
        <div
          class="open-source__projects-item"
        >
          <div
            class="open-source__project open-source__project-id-match"
          >
            <a
              class="open-source__project-link"
              href="https://github.com/maddevsio/idmatch"
              target="_blank"
            >
              <div
                class="open-source__content-wrap"
              >
                <div
                  class="open-source__icon open-source__id-match"
                />
              </div>
               
              <p
                class="open-source__paragraph"
              >
                An open source tool enabling ID cards recognition to ensure security on your premises.
              </p>
            </a>
          </div>
        </div>
        <div
          class="open-source__projects-item"
        >
          <div
            class="open-source__project open-source__project-your-cast"
          >
            <a
              class="open-source__project-link"
              href="https://github.com/maddevsio/yourcast.tv"
              target="_blank"
            >
              <div
                class="open-source__content-wrap"
              >
                <div
                  class="open-source__icon open-source__your-cast"
                />
              </div>
               
              <p
                class="open-source__paragraph"
              >
                A solution to create lists of YouTube videos and stream them online in just a few seconds.
              </p>
            </a>
          </div>
        </div>
        <div
          class="open-source__projects-item"
        >
          <div
            class="open-source__project open-source__project-heimdall"
          >
            <a
              class="open-source__project-link"
              href="https://github.com/maddevsio/heimdall"
              target="_blank"
            >
              <div
                class="open-source__content-wrap"
              >
                <div
                  class="open-source__icon open-source__heimdall"
                />
              </div>
               
              <p
                class="open-source__paragraph"
              >
                A 
                <span>
                  security-monitoring
                </span>
                 solution for Ethereum smart contracts.
              </p>
            </a>
          </div>
        </div>
      </div>
       
      <a
        class="open-source__github-link"
        href="https://github.com/maddevsio"
        target="_blank"
      >
        View more on GitHub
      </a>
    </div>
  </section>
</main>
`;
